lib/rules/attributes-order.js

```diff
@@ -0,0 +1,231 @@
+      const hasAttributesAfterSplattributes =
+        nodeTokens[AttributeType.SPLATTRIBUTES].exists &&
+        nodeTokens[AttributeType.SPLATTRIBUTES].last_index <
+          nodeTokens[AttributeType.ATTRIBUTES].last_index;
+
+      switch (attributeOrder) {
+        case 0:
+          return (
+            nodeTokens[attributeType].last_index >
+              nodeTokens[config.attributeOrder[1]].first_index ||
+            nodeTokens[attributeType].last_index >
+              nodeTokens[config.attributeOrder[2]].first_index ||
+            nodeTokens[attributeType].last_index > nodeTokens[config.attributeOrder[3]].first_index
+          );
+
+        case 2:
+          if (attributeType !== AttributeType.MODIFIERS || !hasAttributesAfterSplattributes) {
+            return (
+              nodeTokens[attributeType].last_index < nodeTokens[config.attributeOrder[1]].last_index
+            );
+          }
+          break;
+
+        case 3:
+          if (attributeType !== AttributeType.MODIFIERS || !hasAttributesAfterSplattributes) {
+            return (
+              nodeTokens[attributeType].last_index < nodeTokens[config.attributeOrder[2]].last_index
+            );
+          }
+          break;
+
+        default:
+          break;
+      }
+    }
+
+    function checkMustacheAttributesOrder(node) {
+      if (!sourceForNode(node)) {
+        return;
+      }
+      const nodeTokens = addMetadata(groupTokens(getMustacheTokens(node, sourceForNode(node))));
+      const attributeType = AttributeType.ATTRIBUTES;
+      if (config.alphabetize && nodeTokens[attributeType].unalphabetized_item_pos >= 0) {
+        const [, source, item] = findUnalphabetizedElement(attributeType, nodeTokens);
+        log({
+          message: createNotAlphabetizedErrorMessage(attributeType, source),
+          line: item.loc.start.line,
+          column: item.loc.start.column,
+          source,
+          node: item,
+        });
+      }
+    }
+
+    function checkElementAttributesOrder(node) {
+      if (!sourceForNode(node)) {
+        return;
+      }
+      const nodeTokens = addMetadata(groupTokens(getTokens(node, sourceForNode(node))));
+
+      for (const attributeType of config.attributeOrder) {
+        if (config.alphabetize && nodeTokens[attributeType].unalphabetized_item_pos >= 0) {
+          const [, source, item] = findUnalphabetizedElement(attributeType, nodeTokens);
+          log({
+            message: createNotAlphabetizedErrorMessage(attributeType, source),
+            line: item.loc.start.line,
+            column: item.loc.start.column,
+            source,
+            node: item,
+          });
+        }
+        if (isAttributeUnordered(attributeType, nodeTokens)) {
+          const [, source, item] = findCurrentElement(attributeType, nodeTokens);
+          log({
+            message: createUnorderedErrorMessage(attributeType, source),
+            line: item.loc.start.line,
```

Comment:
For all of these log statements: When you specify the current `node` to display the violation on, I don't think we should need to include `line`, `column`, `source`.